custom/icds/tasks/ccz_hosting.py

```diff
@@ -28,4 +30,4 @@ def setup_ccz_file_for_hosting(ccz_hosting_id):
         except Exception as e:
             # delete the file from blob db if it was added but later failed
             ccz_utility.remove_file_from_blobdb()
-            raise e
+            six.reraise(*sys.exc_info())
```

Comment:
It is important to capture `sys.exc_info()` immediately inside the `except` block since `ccz_hosting.delete_ccz()` might change the exception context. Example:
```py
>>> import sys
>>> import six

>>> # bad re-raise exception
... try:
...     raise Exception("should re-raise")
... except:
...     try:
...         raise Exception("should NOT re-raise")
...     except:
...         pass
...     six.reraise(*sys.exc_info())
...
Traceback (most recent call last):
  File "<stdin>", line 9, in <module>
  File "<stdin>", line 6, in <module>
Exception: should NOT re-raise

>>> # good re-raise exception
... try:
...     raise Exception("should re-raise")
... except:
...     exc = sys.exc_info()
...     try:
...         raise Exception("should NOT re-raise")
...     except:
...         pass
...     six.reraise(*exc)
...
Traceback (most recent call last):
  File "<stdin>", line 10, in <module>
  File "<stdin>", line 3, in <module>
Exception: should re-raise
```

Reply:
https://github.com/dimagi/commcare-hq/pull/24129/commits/f80a2641320dab474b2e70cc1c59a30d571c3ef6